okhttp/src/main/java/okhttp3/internal/platform/Platform.kt

```diff
@@ -198,7 +198,8 @@ open class Platform {
 
     private val isBouncyCastlePreferred: Boolean
       get() {
-        return System.getProperty("okhttp.platform") == "bouncycastle"
+        val preferredProvider = Security.getProviders()[0].name
```

Comment:
Need registration of providers, not just system property.

cc @swankjesse this was deliberate originally and this is a behaviour change.

Reply:
I like this behavior better